### PR TITLE
Add async packet provider

### PR DESCRIPTION
**Pull request by Shredder121**

### CODE REVIEW

File: LavalinkServer/src/main/java/lavalink/server/Launcher.java [hunk 1]
```diff
@@ -130,7 +132,9 @@ public static void main(String[] args) {
                 log.info("Using default buffer");
                 nativeAudioSendFactory = new NativeAudioSendFactory();
             }
-            AudioConnection.setAudioSendFactory(nativeAudioSendFactory);
+            AudioConnection.setAudioSendFactory(
+                    AsyncPacketProviderFactory.adapt(nativeAudioSendFactory)
+            );
             log.info("JDA-NAS supported system detected. Enabled native audio sending.");
         } else {
             log.warn("This system and architecture appears to not support native audio sending! "
```
Comment (Shredder121) [inline LavalinkServer/src/main/java/lavalink/server/Launcher.java:140]: We could also wrap the original audio send factory in the async wrapper?
So the normal audio sending logic is also non blocking.

Comment (freyacodes) [reply LavalinkServer/src/main/java/lavalink/server/Launcher.java:140]: I'm not sure how that works

**Review by schnapster — APPROVED** [review]
lgtm

**Comment by freyacodes** [comment]
Still doesn't look like sending is fast enough beyond ~560 players

**Pull request merged**